model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -41,8 +42,12 @@ def is_python(backend) -> bool:
     return is_embedded(backend) or is_roundtrip(backend)
 
 
+def is_dace(backend) -> bool:
+    return backend.name.startswith("run_dace_") if backend else False
+
+
 def is_embedded(backend) -> bool:
-    return backend is None
+    return backend in (None, "embedded")
```

Comment:
This function extended its scope from `backend: Backend | None` to `backend: Backend | None | str` and therefore behaves different to `is_dace`. Probably you could resolve the str->backend earlier, then the signature can stay the same.

Reply:
I will revert this to its original form and add a line in the match_marker to go from "embedded" to None

Reply:
From `str` to `Optional[Backend]`!